### PR TITLE
Allow multiple boards on one page

### PR DESCRIPTION
**Pull request by stinnux**
I created a plugin (https://github.com/stinnux/kanboard-bigboard) that allows displaying multiple kanboards/projects on a single page. 

To enable the plugin various functions need to be changed, so that the boardId/projectId can be passed on. 

Also Added Hooks for project list menu

### CODE REVIEW

File: assets/js/src/BoardPolling.js [hunk 1]
```diff
@@ -16,18 +15,23 @@ Kanboard.BoardPolling.prototype.check = function() {
...
-            cache: false,
-            url: $("#board").data("check-url"),
-            statusCode: {
-                200: function(data) {
-                    self.app.get("BoardDragAndDrop").refresh(data);
-                },
-                304: function () {
-                    self.app.hideLoadingIcon();
+        // Poll every board
+        $("table[id=board]").each(function() {
```
Comment (fguillot) [inline assets/js/src/BoardPolling.js]: You should not have multiple HTML elements with the same ID.

File: assets/js/src/BoardPolling.js [hunk 2]
```diff
@@ -16,18 +15,23 @@ Kanboard.BoardPolling.prototype.check = function() {
...
+          var url = $(this).attr("data-check-url");
+            $.ajax({
+                cache: false,
+                url: url,
+                statusCode: {
+                    200: function(data) {
+                        self.app.get("BoardDragAndDrop").refresh(boardid,data);
+                    },
+                    304: function () {
+                        // TODO: Hide loading Icon after all boards have been checked.
```
Comment (fguillot) [inline assets/js/src/BoardPolling.js]: Do not send "work in progress" code or mark the PR as WIP.

File: app/Template/project_list/show.php [hunk 3]
```diff
@@ -1,12 +1,14 @@
 <section id="main">
     <div class="page-header">
         <ul>
+            <?= $this->hook->render('template:projectlist:menubefore', array('project' => $project)) ?>
```
Comment (fguillot) [inline app/Template/project_list/show.php]: You can send these changes in a separate PR to be merged quickly, do not forget to update the documentation if you add new hooks.

File: assets/js/src/BoardPolling.js (hunk 1 above)
Comment (stinnux) [inline assets/js/src/BoardPolling.js]: I've changed the id of the table to include the project id

File: assets/js/src/BoardPolling.js (hunk 2 above)
Comment (stinnux) [inline assets/js/src/BoardPolling.js]: The loading icon is now shown as long as polls are in progress.

File: app/Template/project_list/show.php (hunk 3 above)
Comment (stinnux) [inline app/Template/project_list/show.php]: I've moved the hook to a separate PR and updated the documentation.

**Comment by stinnux** [comment]
Pleaes review the PR again.

File: assets/js/src/BoardDragAndDrop.js [hunk 4]
```diff
@@ -4,72 +4,80 @@ Kanboard.BoardDragAndDrop = function(app) {
...
+
+                task.removeClass("draggable-item-selected");
+
+                if (newColumnId != taskColumnId || newSwimlaneId != taskSwimlaneId || newPosition != taskPosition) {
+                    self.changeTaskState(boardId, taskId);
+                    self.save(saveURL, boardId, taskId, newColumnId, newPosition, newSwimlaneId);
+                }
+            },
+            start: function(event, ui) {
+                console.log("draggable start");
```
Comment (fguillot) [inline assets/js/src/BoardDragAndDrop.js]: You should remove the debug statements

**Comment by fguillot** [comment]
Do not forget to update the CSS, otherwise that introduce breaking changes: https://github.com/kanboard/kanboard/blob/master/assets/sass/_board.sass#L13

**Comment by stinnux** [comment]
I have corrected the css notation.

**Comment by stinnux** [comment]
And removed the console logging

File: app/Model/SubtaskTimeTrackingModel.php [hunk 5]
```diff
@@ -87,6 +87,8 @@ public function getTaskQuery($task_id)
                         self::TABLE.'.start',
                         self::TABLE.'.time_spent',
                         self::TABLE.'.user_id',
+                        self::TABLE.'.is_billable',
```
Comment (fguillot) [inline app/Model/SubtaskTimeTrackingModel.php]: What is it?

File: data/debug.sql [hunk 6]
```diff
@@ -0,0 +1,22 @@
+SELECT count(*) AS _count,
+       subtask_time_tracking.is_billable AS `subtask_time_tracking.is_billable`,
+       subtask_time_tracking.subtask_id AS `subtask_time_tracking.subtask_id`,
```
Comment (fguillot) [inline data/debug.sql]: This file should not be here.

Comment (stinnux) [inline data/debug.sql]: you are right, has been removed.

File: app/Model/SubtaskTimeTrackingModel.php (hunk 5 above)
Comment (stinnux) [inline app/Model/SubtaskTimeTrackingModel.php]: these are fields from a different plugin that have been slipped in. i have removed them.

**Comment by stinnux** [comment]
I don't know what happened here. I corrected BoardPolling.js which has been added incorrectly. However, there seems to be still a conflict. How can i find out which file is in conflict?

**Comment by stinnux** [comment]
I have created a new PR #2624 which is a lot cleaner and should be easily mergable
